w3-aggregation.md

```diff
@@ -0,0 +1,359 @@
+
+```mermaid
+sequenceDiagram
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Storefront->>Authority: invoke `aggregate/offer`
+    Note left of Authority: Request offer to be queued
+    Authority-->>Storefront: receipt issued
+```
+
+### Broker queues the offer
+
+Once broker successfully gets an offer, the offer gets queued for review. A receipt is created to proof the transition of `aggregate/offer` state from `null` into `queued`.
+
+This receipt MUST contain a follow up task in the (`.fx.join` field) that is run when submitted request is processed. It MAY succeed (if aggregate was accepted) or fail (if aggregated was determined to be invalid). The result of the subsequent task CAN be looked up using its receipt.
+
+### Broker reviews and handles the offer
+
+When a broker dequeues the offer from the queue, the offer details MUST be retrievable. With the offer details, the broker will interact with available Filecoin Storage Providers, in order to establish a previously determined number of deals. Depending on storage providers availability, as well as the content present in the offer, the aggregate MAY be handled or not. A receipt is created to proof the transition of `aggregate/offer` state from `queued` into `accepted` or `denied`.
+
+```mermaid
+sequenceDiagram
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Note right of Storefront: Review and handle offer async
+    Authority-->>Storefront: receipt issued
+```
+
+Once offer is accepted broker takes care of arranging and renewing deals.
+
+### Storefront can query state of the aggregate deals
+
+Storefront users MAY want to check details about deals from the content they previously stored. These deals will change over time as they get renewed. Therefore, Storefront should invoke `aggregate/get` capability to gather information about given aggregate identifier.
+
+```mermaid
+sequenceDiagram
+participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Storefront->>Authority: invoke `aggregate/get`
+```
+
+## Capabilities
+
+This section describes the capabilities that form the w3 aggregation protocol, along with the details relevant for invoking capabilities with a service provider.
+
+In this document, we will be looking at `spade.storage` as an implementer of the `aggregate/*` and `offer/*` protocol.
+
+### `aggregate/offer`
+
+A Storefront principal can invoke a capabilty to submit an aggregate ready for offers.
```

Comment:
```suggestion
A Storefront principal can invoke a capabilty to submit an aggregate ready for offers.

```iplsch
type AggregateOffer union {
  Aggregate  "aggregate/offer"
} representation inline {
  discriminantKey "can"
}

type Offer struct {
  with   StorefrontDID
  offer  { CARLink: OfferDetails }
}

type struct OfferDetails {
    size        Int
    commP       Bytes
    src         [URL]
}

type CARLink string
```

I think adding a schema here would be very helpful. Note that I have inlined details as per our last call. Also instead of a list I've used map keyed by CAR cid, which I think is better as it's more deterministic (agnostic of order).